Userland/Libraries/LibELF/DynamicLoader.cpp

```diff
@@ -654,10 +655,13 @@ DynamicLoader::RelocationResult DynamicLoader::do_direct_relocation(DynamicObjec
         auto [dynamic_object_of_symbol, symbol_value] = maybe_resolution.value();
 
         size_t addend = relocation.addend_used() ? relocation.addend() : *patch_ptr;
-        *patch_ptr = addend + dynamic_object_of_symbol.tls_offset().value() + symbol_value;
+        *patch_ptr = addend + dynamic_object_of_symbol.tls_offset().value() + symbol_value + TLS_TP_STATIC_TLS_BLOCK_OFFSET;
 
+#if ARCH(X86_64)
         // At offset 0 there's the thread's ThreadSpecificData structure, we don't want to collide with it.
         VERIFY(static_cast<ssize_t>(*patch_ptr) < 0);
+#endif
```

Comment:
same question here, `if constexpr (TLS_VARIANT == 2) {` ?

Reply:
I also added a similar check when using TLS variant I. Is this better now?